driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/read.bpf.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2022 The Falco Authors.
+ *
+ * This file is dual licensed under either the MIT or GPL 2. See MIT.txt
+ * or GPL2.txt for full copies of the license.
+ */
+
+#include <helpers/interfaces/fixed_size_event.h>
+#include <helpers/interfaces/variable_size_event.h>
+
+/*=============================== ENTER EVENT ===========================*/
+
+SEC("tp_btf/sys_enter")
+int BPF_PROG(read_e,
+	     struct pt_regs *regs,
+	     long id)
+{
+	struct ringbuf_struct ringbuf;
+	if(!ringbuf__reserve_space(&ringbuf, READ_E_SIZE))
+	{
+		return 0;
+	}
+
+	ringbuf__store_event_header(&ringbuf, PPME_SYSCALL_READ_E);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: fd (type: PT_FD) */
+	s32 socket_fd = (s32)extract__syscall_argument(regs, 0);
+	ringbuf__store_s64(&ringbuf, (s64)socket_fd);
+
+	/* Parameter 2: size (type: PT_UINT32) */
+	u32 size = (u32)extract__syscall_argument(regs, 2);
+	ringbuf__store_u32(&ringbuf, size);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	ringbuf__submit_event(&ringbuf);
+
+	return 0;
+}
+
+/*=============================== ENTER EVENT ===========================*/
+
+/*=============================== EXIT EVENT ===========================*/
+
+SEC("tp_btf/sys_exit")
+int BPF_PROG(read_x,
+	     struct pt_regs *regs,
+	     long ret)
+{
+	struct auxiliary_map *auxmap = auxmap__get();
+	if(!auxmap)
+	{
+		return 0;
+	}
+
+	auxmap__preload_event_header(auxmap, PPME_SYSCALL_READ_X);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: res (type: PT_ERRNO) */
+	auxmap__store_s64_param(auxmap, ret);
+
+	if(ret > 0)
+	{
+		/* We read the minimum between `snaplen` and what we really
+		 * have in the buffer.
+		 */
+		unsigned long bytes_to_read = maps__get_snaplen();
```

Comment:
Is snaplen relevant for the `read` syscall?

Reply:
Also, not sure to add a specific snaplen related test case similar like in `sendto`.

Reply:
yeah, we apply the `snaplen` logic to all syscalls that could handle huge numbers of bytes, so `read`, `write`, `readv`, `writev`, `sendmsg`, `sendto` and all the others